addons/stock/models/barcode.py

```diff
@@ -8,8 +8,13 @@ class BarcodeRule(models.Model):
     _inherit = 'barcode.rule'
 
     type = fields.Selection(selection_add=[
-            ('weight', 'Weighted Product'),
-            ('location', 'Location'),
-            ('lot', 'Lot'),
-            ('package', 'Package')
-        ])
+        ('weight', 'Weighted Product'),
+        ('location', 'Location'),
+        ('lot', 'Lot'),
+        ('package', 'Package')
+    ], ondelete={
+        'weight': 'set default',
+        'location': 'set default',
+        'lot': 'set default',
+        'package': 'set default',
```

Comment:
Did stock say to do that, rather than e.g. delete the rule? Because it seems a bit weird that we'd just convert stock barcode rules to "unit product"

Reply:
I invited them to review the code and they didn't seem to object to this particular change, but maybe you're right.

cc @sle-odoo